src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -80,42 +128,207 @@ const SKUpdate& SKNMEA2000Parser::parse128259(const SKSourceInput& input, const
+  unsigned char sid;
+  double windSpeed = N2kDoubleNA;     // m/s
+  double windAngle = N2kDoubleNA;     // in Rad
+  tN2kWindReference windReference;
+
+  if (ParseN2kPGN130306(msg,sid,windSpeed,windAngle,windReference)) {
+    SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+
+    if (!N2kIsNA(windAngle) && !N2kIsNA(windSpeed)) {
+      if (windReference == N2kWind_True_North) {
+        // Ground Wind Speed
+        update->setEnvironmentWindSpeedOverGround(windSpeed);
+        // Ground Wind Direction
+        update->setEnvironmentWindDirectionTrue(windAngle);
+      }
+      else if (windReference == N2kWind_Magnetic) {
+        // Ground Wind Speed
+        update->setEnvironmentWindSpeedOverGround(windSpeed);
+        // Ground Wind Direction refered to magnetic north
+        update->setEnvironmentWindDirectionMagnetic(windAngle);
+      }
+      //TODO: correct if Timos Typo will be corrected in library
+      else if (windReference == N2kWind_Apprent) {
+        // AWS Apparent Wind Speed
+        update->setEnvironmentWindSpeedApparent(windSpeed);
+        // AWA pos coming from starboard, neg from port, relative to centerline vessel
+        if (windAngle > M_PI) windAngle *= -1;
+        update->setEnvironmentWindAngleApparent(windAngle);
+      }
+      else if (windReference == N2kWind_True_boat) {
+        // Ground Wind
+        update->setEnvironmentWindSpeedTrue(windSpeed);
+        // Ground Wind +/- starboard/port
+        if (windAngle >M_PI) windAngle *= -1;
+        update->setEnvironmentWindAngleTrueGround(windAngle);
+      }
+      else if (windReference == N2kWind_True_water) {
+        // TWS (water refered) True "Sailing" Wind
+        update->setEnvironmentWindSpeedTrue(windSpeed);
+        // TWA (water refered) +/- starboard/port
+        if (windAngle >M_PI) windAngle *= -1;
+        update->setEnvironmentWindAngleTrueWater(windAngle);
+      }
     }
 
     _sku = update;
     return *_sku;
   }
   else {
-    INFO("Unable to parse N2kMsg with PGN %i", msg.PGN);
+    DEBUG("Unable to parse NMEA2000 with PGN %i", msg.PGN);
     return _invalidSku;
   }
 }
+
+//  ***********************************************
+//   PGN 127245 Rudder
+//        →  sid
+//        →  RudderPosition [rad]
+// *  ********************************************** */
+const SKUpdate& SKNMEA2000Parser::parse127245(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char instance;
+  tN2kRudderDirectionOrder rudderDirectionOrder;
+  double rudderPosition = N2kDoubleNA;
+  double angleOrder = N2kDoubleNA;
+
+  if (ParseN2kRudder(msg,rudderPosition,instance,rudderDirectionOrder,angleOrder)) {
+    // validation check max +/- 45°
+    if (!N2kIsNA(rudderPosition) && (rudderPosition <= M_PI_4) && (rudderPosition >= M_PI_4) ) {
```

Comment:
Your condition says that `rudderPosition` must be less than 45° and more than 45°at the same time. That is not possible.

This is why the test does not pass (and why tests are so awesome!!!).